lib/livebook/zta/cloudflare.ex

```diff
@@ -8,63 +8,50 @@ defmodule Livebook.ZTA.Cloudflare do
   @assertion "cf-access-jwt-assertion"
   @renew_afer 24 * 60 * 60 * 1000
 
-  defstruct [:name, :req_options, :identity]
+  defstruct [:name, :req_options, :identity, :keys]
 
   def start_link(opts) do
     identity = identity(opts[:identity][:key])
-    options = [req_options: [url: identity.certs], identity: identity]
+    options = [req_options: [url: identity.certs], identity: identity, keys: nil]
     GenServer.start_link(__MODULE__, options, name: opts[:name])
   end
 
-  def authenticate(name, conn) do
+  def authenticate(name, conn, fields: fields) do
     token = get_req_header(conn, @assertion)
-    GenServer.call(name, {:authenticate, token})
+    user = GenServer.call(name, {:authenticate, token, fields})
+    if user, do: {put_session(conn, :current_user_id, user.id), user}, else: {conn, nil}
```

Comment:
For the ZTA we don't access `:current_user_id`, so probably doesn't make sense to put it in the session either?

Reply:
We access it in the `user_hook`. I reverted the last change.

Reply:
@josevalim I'm missing something, is there a reason we expect each provider to set `current_user_id` instead of relying on identity_data?

Reply:
I think we can follow the same idea as now but using only `identity_data`